### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'datesList[_this.state.numVisibleDays - 1].date')

### ISSUE THREAD

**Issue by gustavomanolo** [BugiDev/react-native-calendar-strip#303]
HI, i've added a calendar strip instance like this:

`<CalendarStrip
        style={{ height: 100, paddingTop: 10, paddingBottom: 10 }}
        minDate={moment()}
        maxDate={moment().add(10, "days")}
        datesBlacklist={(date) => {
          return (date.isoWeekday() === 6 || date.isoWeekday() === 7);
        }}
      />`

But i'm getting an error:
TypeError: undefined is not an object (evaluating 'datesList[_this.state.numVisibleDays - 1].date')

If i remove the "datesBlacklist" property, it works but i need to disable saturdays and sundays.

I'm using version => "react-native-calendar-strip": "^2.2.3"

Thanks in advance.

**Comment by peacechen** [comment]
Are you passing in another other props?  Does your code work with the sample app?
https://github.com/BugiDev/react-native-calendar-strip#development-with-sample-application

**Comment by phatmann** [comment]
This error occurs when `componentDidUpdate` is called before `onLayoutDebounce` is called via a timer. `onLayoutDebounce` is where the `numVisibleDays` state is set.

**Comment by phatmann** [comment]
I found a fix. In componentDidUpdate, replace this line

       prevProps.datesWhitelist !== this.props.datesWhitelist  ||

with this line

        JSON.stringify(prevProps.datesWhitelist) !== JSON.stringify(this.props.datesWhitelist) ||

In other words, the `prevProps.datesWhitelist` reference does not equal `this.props.datesWhitelist`, even though the property has not changed. I do not understand why that would be.

This fix would be done for all the property comparisons.

The pros and cons of this fix are pretty obvious. `JSON.stringify` is better in that two equal arrays will not trigger an update, but it is less performant for large arrays.

**Comment by phatmann** [comment]
Here is a possible fix: stop componentDidUpdate from being called until layout is done. I places these lines at beginning of componentDidUpdate and the problem went away:

        if (!this.state.dayComponentWidth) {
          return
        }

I am not sure if this will cause other problems or not, which is why I will not submit a PR with this fix yet.

**Comment by mtimofiiv** [comment]
@phatmann I made a PR out of your suggested fix, which also worked for me. https://github.com/BugiDev/react-native-calendar-strip/pull/314

**Comment by peacechen** [comment]
Thanks @mtimofiiv and @phatmann for identifying the issue.

That change will likely cause undesirable side effects, and is a band-aid for the root problem. Please revert that change in the PR and instead add this to the state initializer:
https://github.com/BugiDev/react-native-calendar-strip/blob/master/src/CalendarStrip.js#L142

```
    this.state = {
      startingDate,
      selectedDate,
      datesList: [],
      dayComponentWidth: 0,
      height: 0,
      monthFontSize: 0,
      selectorSize: 0,
      numVisibleDays: props.numDaysInWeek,
    };
```

**Comment by mtimofiiv** [comment]
@peacechen you're right. I've updated the PR by reverting the old change and adding the initial state as you suggested.

**Comment by peacechen** [comment]
Thanks again @mtimofiiv for updating the PR.  The fix has been published in 2.2.4

**Comment by jakehobbs** [comment]
I am still having this issue is 2.2.4.

**Comment by mtimofiiv** [comment]
Hmm, the fix has not seemed to work, still getting the same error on `2.2.4`. I can look into this a bit further later.

**Comment by phatmann** [comment]
This fix caused other issues for me. I can look into it if needed.

**Comment by Marcoo09** [comment]
In 2.2.5 I have not this problem anymore
